software/19.Socket.io/README.md

```diff
@@ -0,0 +1,131 @@
+
+#### Ressources :
+- [Class](https://www.typescriptlang.org/docs/handbook/2/classes.html)
+- [Base Affichage](https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html)
+
+## Step 04 :
+
+It is time to get information on your input.
+
+For that you need to create a function that achieve a query on the id of your "input" once this one is submit. 
+
+Once you get the message, you need to get the message of your server to print it on your shell.
+
+#### Ressources :
+- [Socket listener](https://socket.io/docs/v4/index.html)
+
+## Step 05 :
+
+Having our message on your shell is great but receive your message in your browser console is better.
+
+Our message from input are, for now, send to our server; We need to aks it to retransmit it with the aim of print it in the browser console.
+
+Creat a function who get back the message nearby the server to print it in the browser console.
+#### Ressources :
+- [Socket Client](https://socket.io/docs/v4/client-socket-instance/)
+
+## Step 06 :
+
+With this step, you will send a message to all the user on the page HTML each time there is a connection or disconnection.
+
+For this end, we will creat a function that add contents with the help of a query on our HTML page.
+You have to take in consideration the message that your server display to sent it to your front.
+
+To test your code, we will open two pages of your browser with the url : http://localhost:8080/
+
+Throughout the connection, you have to print "Connection" and "Disconnexion" when you leave the page.
+
+#### Ressources :
+
+- [Emettre une socket](https://socket.io/docs/v3/emit-cheatsheet/index.html)
+- [Requête HTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector)
+
+## Step 07 :
+
+Creat a function that sent the content of our input on our HTML page.
+
+
+For this, we will create a function that achieve a query on our HTML page to add it the message of the input.
+
+#### Ressources :
+
+- [Release a socket] (https://socket.io/docs/v3/emit-cheatsheet/index.html)
+- [Request HTML] (https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector)
+- [Socket Client](https://socket.io/docs/v4/client-socket-instance/)
+## Step 08 :
+
+It is time to identify the message more clearly. 
+Creat an id for every user to allow in order to receive messages of users with their name or a determinant.
+
+Exemple:
+
+You: "message_sent"
+Tony: "message_receive"
+
+## Step 09 :
+
+For this last step we need tro add the date and the hour of when the message was sent.
+
+## Bonus
+
+* Creat rooms that allow you to send a message to every user that are in the same room as you.
+* Amelioration of your Front with React.
+* Authentification of the user with socket.
+* Add a data base to stock the message et refresh when there is a connexion.
+
+#### Ressources:
+- [React](https://reactjs.org/)
+- [Room Socket](https://socket.io/docs/v3/rooms/index.html)
+- [Authentifaciton Socket](https://socket.io/docs/v3/middlewares/)
```

Comment:
You missed authors and organization.
See [other workshops](https://github.com/PoCInnovation/Workshops/tree/master/software/) for details